### ISSUE TITLE
Cannot filter array without keys

### ISSUE THREAD

**Issue by newPOPE** [particle-php/Filter#63]
### What

```php
$foo = [
  ['name' => 'john'],
  ['name' => 'daisy'],
];

$filter = new Filter;
$filter->all()->each(function(Filter $filter) use ($translateFnc) {
  $filter->value('name')->callback($translateFnc);
});
```

### What is expected
 
Filter doesn't touch `name` values in collection. Is there what I am doing wrong or there is a bug?.

**Comment by rick-nu** [comment]
```php
$foo = [
    'names' => [
        ['name' => 'john'],
        ['name' => 'daisy'],
    ]
];

$filter = new Filter;
$filter->value('names')->each(function(Filter $filter) use ($translateFnc) {
  $filter->value('name')->callback($translateFnc);
});
```

Hi @newPOPE, try the above example. 

`->all()` targets all keys of an array. Since you have no named keys, the Filter doesn't know what to do. Let me know if this solved your question

**Comment by newPOPE** [comment]
Hello, nope this is not good for me (I expected `->all()` takes all items because key exists everytime, number, string, ...). I using this filter for translations in structs and example looks like this:

```php
$messages = [/*array of some arrays/messages in app without explicit keys*/];
// and I must use filter like this :/, it works but...
[
  ... // some other properties

  // $factory returns filter for filtering message struct
  'messages' => $translationFilterFactory->create(MessagesTranslationFilter::class)->filter(['items' => $messages])['items'],
]
```

Is there any reason why `->all()` doesn't contains items "without" keys?

**Comment by rick-nu** [comment]
The internals of Filter build a "chain" of "filter rules" for every array key:
https://github.com/particle-php/Filter/blob/master/src/FilterResource.php#L322-L343
`->all()` does expect explicit keys to build those "filter rune chains" properly.

Looking at the internals, the keys that are used now are `0` and `1`. Using `->all()` you target the the values for those keys, which are arrays, not values. Hacking around that you use an `each`, which does expect it te be an array with multiple arrays in it. But since it's not an array containing new arrays, that filters won't able to target anything.

So long story short, `Filter` doesn't work like that :innocent: 
If we allowed it to work like that, it would be dirty/hacky. And that's pretty much the whole point of Particle, be be short but clear/explicit.
